src/methods/geom_relations/common.jl

```diff
@@ -0,0 +1,28 @@
+# Identical boilerplate methods for geom relations live here
+
+for f in (:coveredby, :crosses, :disjoint, :overlaps, :touches, :within)
+    _f = Symbol(:_, f)
+
+    # Features
+    @eval begin
+        $_f(::FeatureTrait, g1, ::Any, g2) = $f(geometry(g1), g2)
+        $_f(::Any, g1, ::FeatureTrait, g2) = $f(g1, geometry(g2))
+        $_f(::FeatureTrait, g1, ::FeatureTrait, g2) = $f(geometry(g1), geometry(g2))
+    end
+
+    # Table rows
+    @eval begin
+        $_f(::Nothing, g1, ::Any, g2) = $f(_geometry_or_error(g1), g2)
+        $_f(::Any, g1, ::Nothing, g2) = $f(g1, _geometry_or_error(g2))
+        $_f(::Nothing, g1, ::Nothing, g2) = 
+            $f(_geometry_or_error(g1), _geometry_or_error(g2))
```

Comment:
~This will probably mess with extents.  So we'll have to make sure dispatch on those is more specific.~

But maybe we could have a geometrycolumn kwarg here?  That gets passed to _geometry_or_error?

For the two arg case we can tell the user to figure it out

Reply:
Ok, handled all the ambiguities and added the kw